2d object recognition yolov5 v1

kvnptl opened the pull request:
<!--- Make the title descriptive! Provide a general summary of your changes in the title above -->

## Changelog
<!-- Add a list of bullets with the summary of your changes -->
<!-- Try to use infinitives: Fix, Remove, Rename, Add, Refactor -->

* [Add] YOLOv5 for rgb_object_recognition in PyTorch
* [Add] DGCNN for pc_object_recognition in PyTorch 

## Related PRs
<!-- Mention any other pull requests that need to be merged (and in which order, if applicable). -->
<!--If your PR is related to an issue, you can close the issue by using keywords: https://help.github.com/en/articles/closing-issues-using-keywords -->
<!-- For example, just write: Closes #31 -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code doesn't contain unnecessary comment blocks (e.g. unused code, templates of `package.xml` or `CMakeLists.txt`)
- [x] I have updated the `package.xml` and `CMakeLists.txt` with the correct dependencies.
- [x] I have updated the documentation accordingly.

<!-- Click on the preview button to make sure everything is correctly formatted -->

kvnptl committed:
new: yolov5 integration for rgb object recognition

kvnptl committed:
update: x_offset and y_offset calculation

RavisankarSelvaraju committed:
Noetic (#1)

* update 3d pointcloud recoginition in pytorch

* dgcnn_pytorch_classifier

kvnptl committed:
bug fix: bounding box calculation

kvnptl committed:
update: import paths

kvnptl committed:
update: relative paths

kvnptl committed:
bug_fix: point cloud node

kvnptl committed:
update: add scaling factor

kvnptl committed:
update:

kvnptl committed:
update: change back the gitignore

kvnptl committed:
change: temporarily disabled pc object recognition

sthoduka commented:
Can you look through the files included in this PR and remove the ones that are not needed, and remove commented lines etc.? For DGCNN what do the version numbers in the filename indicate?

kvnptl committed:
[mir_object_recognition] add visualization for rgb filtered point cloud and rgb pose bug fix

- bug fix for rgb pose
- add new min max limits for tower camera
- add publisher for rgb filtered point cloud visualization
- bug fix: add flag for pc recognizer (functionality: don't run or wait for pc when pc
flag is false)
